src/lib/features/client-feature-toggles/cache/client-feature-toggle-cache.ts

```diff
@@ -177,53 +177,42 @@ export class ClientFeatureToggleCache {
 
     private async onUpdateRevisionEvent() {
         if (this.flagResolver.isEnabled('deltaApi')) {
-            await this.pollEvents();
+            await this.listenToRevisionChange();
         }
     }
 
-    public async pollEvents() {
+    public async listenToRevisionChange() {
+        const keys = Object.keys(this.cache);
+
+        if (keys.length === 0) return;
         const latestRevision =
             await this.configurationRevisionService.getMaxRevisionId();
 
-        if (this.currentRevisionId === 0) {
-            await this.populateBaseCache(latestRevision);
-        } else {
-            const changeEvents = await this.eventStore.getRevisionRange(
-                this.currentRevisionId,
-                latestRevision,
-            );
-
-            const changedToggles = [
-                ...new Set(
-                    changeEvents
-                        .filter((event) => event.featureName)
-                        .map((event) => event.featureName!),
-                ),
-            ];
-            const newToggles = await this.getChangedToggles(
-                changedToggles,
-                latestRevision, // TODO: this should come back from the same query to not be out of sync
-            );
-
-            if (this.cache.development) {
-                this.cache.development.addRevision(newToggles);
-            }
-        }
-        this.currentRevisionId = latestRevision;
-    }
+        const changeEvents = await this.eventStore.getRevisionRange(
+            this.currentRevisionId,
+            latestRevision,
+        );
 
-    private async populateBaseCache(latestRevisionId: number) {
-        const features = await this.getClientFeatures({
-            environment: 'development',
-        });
-        if (this.cache.development) {
-            this.cache.development.addRevision({
-                updated: features as any, //impressionData is not on the type but should be
-                removed: [],
-                revisionId: latestRevisionId,
-            });
+        const changedToggles = [
+            ...new Set(
+                changeEvents
+                    .filter((event) => event.featureName)
+                    .map((event) => event.featureName!),
+            ),
+        ];
+        const newToggles = await this.getChangedToggles(
+            changedToggles,
+            latestRevision, // TODO: this should come back from the same query to not be out of sync
```

Comment:
This looks ok, but I was thinking about if we might not want to return the revisionId, from query .

Because the flow is getMaxRevisionId(), based on which we get the events, and if now with those events, go pull data and revision ID has changed, it will get out sync. I think best is

If we getMaxRevisionId, we use that across all the calls we make.